src/Components/ImagesTable/ImagesTableToolbar.tsx

```diff
@@ -60,11 +62,50 @@ const ImagesTableToolbar: React.FC<imagesTableToolbarProps> = ({
 }: imagesTableToolbarProps) => {
   const [showDeleteModal, setShowDeleteModal] = useState(false);
   const [showDiffModal, setShowDiffModal] = useState(false);
+  const [isLintExp, setIsLintExp] = React.useState(true);
+  const onToggleLintExp = (_event: React.MouseEvent, isExpanded: boolean) => {
+    setIsLintExp(isExpanded);
+  };
+  const [warningExpanded, setWarningExpanded] = useState(true);
+
+  const dispatch = useAppDispatch();
+
   const selectedBlueprintId = useAppSelector(selectSelectedBlueprintId);
   const blueprintSearchInput = useAppSelector(selectBlueprintSearchInput);
   const blueprintVersionFilterAPI = useAppSelector(
     selectBlueprintVersionFilterAPI,
   );
+  const storedWarnings = useAppSelector(
+    selectWarningsContentForSelectedBlueprint,
+  );
+
+  const { data: blueprintDetails } = useGetBlueprintQuery(
+    { id: selectedBlueprintId! },
+    { skip: !selectedBlueprintId },
+  );
+
+  const { trigger: fixupBlueprint } = useFixupBlueprintMutation();
+
+  const lintWarnings = React.useMemo(
+    () => blueprintDetails?.lint?.warnings ?? [],
+    [blueprintDetails],
+  );
+
+  React.useEffect(() => {
+    if (selectedBlueprintId && blueprintDetails !== undefined) {
+      dispatch(
+        setWarningsContent({
```

Comment:
Note that I'm a React/Redux noob so please forgive me if this question is utterly stupid. Why do we need to store the warnings in the redux store? Cannot we just retrieve them from the blueprint details that we already fetched, and render them? I don't understand the need to keep a second copy in the local state.